src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

```diff
@@ -400,10 +402,367 @@ struct npc_instructor_zhi : public ScriptedAI
+
+    // Spells
+    SPELL_MEDITATION_TIMER_BAR      = 116421,
+    SPELL_CAVE_OF_SCROLLS_CREDIT    = 102447,
+    SPELL_AYSA_CAVE_OF_SCROLLS_COMP = 128598,
+    SPELL_SUMMON_MASTER_LI_FEI      = 102445,
+    SPELL_SUMMON_AMBERLEAF_TROUBLEMAKER = 114710,
+
+    // Actions
+    ACTION_START_MEDITATION  = 1,
+    ACTION_FINISH_MEDITATION = 2,
+
+    // Events
+    EVENT_SUMMON_AMBERLEAF = 1,
+
+    // Texts
+    SAY_AYSA_HELP                   = 0,
+    SAY_AYSA_FINISH_MEDITATION      = 1
+};
+
+// 7756 - Areatrigger
+class at_cave_of_meditation : public AreaTriggerScript
+{
+public:
+    at_cave_of_meditation() : AreaTriggerScript("at_cave_of_meditation") { }
+
+    bool OnTrigger(Player* player, AreaTriggerEntry const* /*areaTrigger*/) override
+    {
+        if (player->IsAlive() && player->GetQuestStatus(QUEST_THE_WAY_OF_THE_TUSHUI) == QUEST_STATUS_INCOMPLETE)
+        {
+            if (!player->HasAura(SPELL_MEDITATION_TIMER_BAR))
+                player->CastSpell(player, SPELL_MEDITATION_TIMER_BAR, TRIGGERED_FULL_MASK);
+
+            return true;
+        }
+        return false;
+    }
+
+    bool OnExit(Player* player, AreaTriggerEntry const* /*trigger*/) override
+    {
+        if (player->HasAura(SPELL_MEDITATION_TIMER_BAR))
+            player->RemoveAura(SPELL_MEDITATION_TIMER_BAR);
+
+        return true;
+    }
+};
+
+// 7645 - Areatrigger
+class at_inside_of_cave_of_meditation : public AreaTriggerScript
+{
+public:
+    at_inside_of_cave_of_meditation() : AreaTriggerScript("at_inside_of_cave_of_meditation") { }
+
+    bool OnTrigger(Player* player, AreaTriggerEntry const* /*areaTrigger*/) override
+    {
+        if (player->IsAlive() && player->GetQuestStatus(QUEST_THE_WAY_OF_THE_TUSHUI) == QUEST_STATUS_INCOMPLETE)
+        {
+            if (!player->HasAura(SPELL_SUMMON_MASTER_LI_FEI))
+            {
+                Creature* aysa = player->FindNearestCreatureWithOptions(40.0f, { .StringId = "npc_aysa_quest_29414" });
+
+                if (!aysa)
+                    return false;
+
+                aysa->AI()->Talk(SAY_AYSA_HELP, player);
+                aysa->AI()->DoAction(ACTION_START_MEDITATION);
+                player->CastSpell(player, SPELL_SUMMON_MASTER_LI_FEI, TRIGGERED_FULL_MASK);
+            }
+
+            return true;
+        }
+        return false;
+    }
+
+    bool OnExit(Player* player, AreaTriggerEntry const* /*trigger*/) override
+    {
+        if (!talkDone)
+        {
+            if (player->IsAlive() && player->GetQuestStatus(QUEST_THE_WAY_OF_THE_TUSHUI) == QUEST_STATUS_COMPLETE)
+            {
+                Creature* aysa = player->FindNearestCreatureWithOptions(40.0f, { .StringId = "npc_aysa_after_quest_29414" });
+
+                if (!aysa)
+                    return false;
+
+                aysa->AI()->Talk(SAY_AYSA_FINISH_MEDITATION);
+                talkDone = true;
+            }
+        }
+
+        return true;
+    }
+
+private:
+    bool talkDone = false;
```

Comment:
Fundamentally broken, there is only one copy of `at_inside_of_cave_of_meditation` in worldserver, you will need to store it in script of the summoned creature